pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/DSMServer.java

```diff
@@ -250,7 +246,19 @@ public static void main(String[] args) {
         LogUtil.addAppEngineEnvVarsToMDC();
         // respond GAE dispatcher endpoints as soon as possible
         // immediately lock isReady so that ah/start route will wait
-        SparkBootUtil.startSparkServer(null, cfg.getConfig("portal"));
+
+        SparkBootUtil.startSparkServer(new SparkBootUtil.AppEngineShutdown() {
```

Comment:
It feels like all this startup and shutdown stuff would be clearer if broken out of this colossal file.

Reply:
I agree, and this PR has on net removed about 60 lines from DSMServer.  The challenge has been identifying the bits that are common to DSS, DSS, and Housekeeping.